core/src/main/scala/caliban/validation/Validator.scala

```diff
@@ -722,6 +728,65 @@ object Validator {
     }
   }
 
+  private def validateDirectives(types: List[__Type]): IO[ValidationError, Unit] = {
+
+    def validateArguments(args: Map[String, InputValue], errorContext: String): IO[ValidationError, Unit] = {
+      val explanatoryText =
+        s"""The directive argument must not have a name which begins with the characters "__" (two underscores)"""
+      val argumentErrorContextBuilder = (name: String) => s"Argument '$name' of $errorContext"
+      IO.foreach_(args.map(_._1))(argName =>
+        doesNotStartWithUnderscore[String](argName, identity, argumentErrorContextBuilder(argName), explanatoryText)
+      )
+    }
+
+    def validateDirective(directive: Directive, errorContext: String) = {
+      val directiveErrorContext = s"Directive '${directive.name}' of $errorContext"
+      for {
+        _ <- doesNotStartWithUnderscore(directive, directiveErrorContext)
+        _ <- validateArguments(directive.arguments, directiveErrorContext)
+      } yield ()
+    }
+
+    def validateDirectives[T](
+      t: T,
```

Comment:
You pass `T` and `T => Option[List[Directive]]`, why not just pass `Option[List[Directive]]`? Then you won't need the type parameter anymore.

Reply:
done :)